### PR TITLE
Intermediate hashes: split account & storage changes

### PR DESCRIPTION
**Pull request by vorot93**
Need to benchmark for perf improvement.

### CODE REVIEW

**Review by jomuel — APPROVED** [review]
Looks good so far, but if we want to improve performance I expected the `PrefixSet` implementation to be changed too. If the account and storage changes are seperate, they are traversed in order again and we can use an array with index and look up in O(1) instead of using a `BTreeSet`.

**Comment by jomuel** [comment]
@vorot93 Let's merge this and I'll open a second PR with the recent improvements from silkworm soon. The changed `PrefixSet` implementation alone saves about 70% time in the benchmark test with the populated trie tables.

File: src/trie/prefix_set.rs [hunk 1]
```diff
@@ -1,28 +1,66 @@
...
+        Self {
+            keys: Vec::new(),
+            sorted: true,
+            index: 0,
+        }
+    }
+
+    fn sort(&mut self) {
+        self.keys.sort();
+        self.keys.dedup();
```
Comment (vorot93) [inline src/trie/prefix_set.rs:28]: Missing `self.sorted = true;`?

Comment (jomuel) [reply src/trie/prefix_set.rs:28]: Yes, added it. I wonder how it could still be faster without it though.

**Pull request merged**